packages/api/src/external/shared/patient-demographics.ts

```diff
@@ -0,0 +1,211 @@
+import { Address } from "@metriport/core/domain/address";
+import { Contact } from "@metriport/core/domain/contact";
+import { USState } from "@metriport/core/domain/geographic-locations";
+
+export const addressSeparator = "||";
+
+type LinkDemoDataAddress = {
+  line?: string;
+  city?: string;
+  state?: string;
+  zip?: string;
+  country?: string;
+};
+
+export type LinkDemoData = {
+  dob?: string;
+  gender?: string;
+  names: {
+    firstName: string;
+    lastName: string;
+  }[];
+  telephoneNumbers: string[];
+  emails: string[];
+  addresses: LinkDemoDataAddress[];
+  driversLicenses: {
+    value: string;
+    state: string;
+  }[];
+  ssns: string[];
+};
+
+export function scoreLink(
+  patientDemographics: LinkDemoData,
+  linkDemographics: LinkDemoData
+): boolean {
+  let scoreThreshold = 20;
+  let score = 0;
+  // DOB exact match
+  if (patientDemographics.dob === linkDemographics.dob) score += 8;
+  // DOB approximate match
+  if (patientDemographics.dob && linkDemographics.dob) {
+    const patientDobSplit = splitDob(patientDemographics.dob);
+    const linkedDobSplit = splitDob(linkDemographics.dob);
+    if (linkedDobSplit.filter(patientDobSplit.includes).length > 2) score += 2;
+  }
+  // Gender exact match
+  if (patientDemographics.gender && linkDemographics.gender) {
+    if (patientDemographics.gender === linkDemographics.gender) score += 1;
+  }
+  // Names exact match
+  if (linkDemographics.names.some(patientDemographics.names.includes)) score += 10;
+  // Names approximate match
+  // TODO
+  // Address exact match
+  // TODO Mark the first address as current in Dash
+  const overlappingAddresses = linkDemographics.addresses.filter(a =>
+    patientDemographics.addresses.includes(a)
+  );
+  score += 2 * overlappingAddresses.length;
+  const currentPatientAddress = patientDemographics.addresses[0];
+  if (currentPatientAddress) {
+    if (linkDemographics.addresses.map(a => a.city).includes(currentPatientAddress.city))
+      score += 0.5;
+    if (linkDemographics.addresses.map(a => a.zip).includes(currentPatientAddress.zip))
+      score += 0.5;
+  }
+  // Address approximate match
+  // TODO
+  // Telephone exact match
+  const overlappingTelephone = linkDemographics.telephoneNumbers.filter(a =>
+    patientDemographics.telephoneNumbers.includes(a)
+  );
+  score += 2 * overlappingTelephone.length;
+  // Telephone approximate match
+  // TODO
+  // Emails approximate match
+  const overlappinEmail = linkDemographics.emails.filter(a =>
+    patientDemographics.emails.includes(a)
+  );
+  score += 2 * overlappinEmail.length;
+  // SSN exact match
+  if (linkDemographics.ssns.length > 0) scoreThreshold = 21;
+  const overlappingSsn = linkDemographics.ssns.filter(a => patientDemographics.ssns.includes(a));
+  if (overlappingSsn.length > 0) score += 5;
+  // SSN approximate match
+  // TODO
+  return score >= scoreThreshold;
+}
+
+export function patientToLinkedDemoData(patient: Patient): LinkDemoData {
```

Comment:
nit: can function name convey that this is a function that is normalizing the data too?

Reply:
and be explicit in a docstring about what the normalization "protocol" we are following is. I think its important to be overly explicit about this sort of thing since normalization is an important piece of this process. For example, epic does a lot more address normlaization than what we are doing here. 

I would even suggest separating each of name, tele, email, address into its own functions called normalizeTelephone, etc.